ert_gui/tools/plot/plot_api.py

```diff
@@ -95,31 +99,35 @@ def data_for_key(self, case, key):
         except ValueError:
             return data
 
-    def observations_for_obs_keys(self, case, obs_keys):
+    def observations_for_key(self, case, key):
         """Returns a pandas DataFrame with the datapoints for a given observation key for a given case. The row index
         is the realization number, and the column index is a multi-index with (obs_key, index/date, obs_index),
         where index/date is used to relate the observation to the data point it relates to, and obs_index is
         the index for the observation itself"""
         try:
-            measured_data = MeasuredData(
-                self._facade, obs_keys, case_name=case, load_data=False
-            )
-            data = measured_data.data
-        except loader.ObservationError:
-            data = pd.DataFrame()
-        expected_keys = ["OBS", "STD"]
-        if not isinstance(data, pd.DataFrame):
-            raise TypeError(
-                "Invalid type: {}, should be type: {}".format(type(data), pd.DataFrame)
-            )
-        elif not data.empty and not set(expected_keys).issubset(data.index):
-            raise ValueError(
-                "{} should be present in DataFrame index, missing: {}".format(
-                    ["OBS", "STD"], set(expected_keys) - set(data.index)
-                )
-            )
-        else:
-            return data
+            with Storage.session() as client:
+                if self._all_cases is None:
+                    self._get_all_cases()
+                resp = client.get(
+                    f"/ensembles/{self._all_cases[case]}/records/{key}/observations"
+                ).json()
+                obs = resp[0]
+                try:
+                    int(obs["x_axis"][0])
+                    key_index = [int(v) for v in obs["x_axis"]]
+                except ValueError:
+                    key_index = [pd.Timestamp(v) for v in obs["x_axis"]]
+
+                data_stuct = {
```

Comment:
Pedantic warning ⚠️  Should the variable `data_stuct` be renamed to `data_struct` ?